src/search/include/search/synchronous_product.h

```diff
@@ -447,10 +447,13 @@ get_time_successors(const CanonicalABWord<Location, ConstraintSymbolType> &canon
  * Compute the successors by following all transitions in the TA and ATA for one time successor and
  * one symbol.
  * */
-template <typename Location, typename ActionType, typename ConstraintSymbolType>
+template <typename Location,
+          typename ActionType,
+          typename ConstraintSymbolType,
+          bool use_location_constraints = false>
 std::vector<CanonicalABWord<Location, ConstraintSymbolType>>
 get_next_canonical_words(
-  const automata::ta::TimedAutomaton<Location, ConstraintSymbolType> &ta,
+  const automata::ta::TimedAutomaton<Location, ActionType> &ta,
```

Comment:
I am wondering: Is it possible, that the ta actually requires all three template parameters? Locations (to identify locations), ActionType (for label-sync on jumps), and ConstraintSymbolType (for labeling locations)? Or do we abuse LocationT to argue about AP-labels for locations?
Remark: It seems we do. Does this not cause potential problems when the location type and the constraintSymbol type are different?

Reply:
Yes, we actually statically assert that:
https://github.com/morxa/mtlsyn/blob/91ba7285ee7f7f48743785ae30eaeee2db95384f/src/search/include/search/search.h#L115-L116

We could also actually automatically define the type using that, I'll have another look